### ISSUE TITLE
heads up bust didn't end tourney

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#283]
```
Dealing 30000 60000 Four Card Chowaha (High/Low Eight Qualifier)

thumbers blinds 15000
Pokerbrat blinds 30000
2 B > thumbers    95738 15000  [  ] [  ] [  ] [  ]
8     Pokerbrat  119262 30000  [  ] [  ] [  ] [  ]
thumbers calls
Pokerbrat checks
2 B   thumbers    80738 30000  [  ] [  ] [  ] [  ]
8   > Pokerbrat  119262 30000  [  ] [  ] [  ] [  ]
Boards:
Td Qs 5h

3h 5s Ts

5d 8d 8h
Pokerbrat bets 30000
thumbers calls
2 B   thumbers   50738 60000  [  ] [  ] [  ] [  ]
8   > Pokerbrat  89262 60000  [  ] [  ] [  ] [  ]
Boards:
Td Qs 5h
         4s
3h 5s Ts
         7s
5d 8d 8h
Pokerbrat bets 60000
thumbers calls and is all-in
2 B > thumbers       0 110738  [  ] [  ] [  ] [  ]
8     Pokerbrat  29262 120000  [  ] [  ] [  ] [  ]
Boards:
Td Qs 5h
         4s
3h 5s Ts    Jh
         7s
5d 8d 8h
Uncalled 9262 returned to Pokerbrat
Pokerbrat  Kc 7d 9d 8s
     High  Full House (Eights over Sevens)   8d 8h 8s 7s 7d
      Low  Eight (8 7 5 4 3)                 8s 7d 5s 4s 3h
thumbers   Kh 9h 3d Th
     High  King High Straight                Kh Qs Jh Td 9h
      Low  none
Pokerbrat wins 110738 for High
Pokerbrat wins 110738 for Low

thumbers busts in 2nd place, winning: 390
Too few players, pausing...
deadhead: huh
deadhead: Ugh. Well, Pokerbrat wins again.
deadhead: I'll fix the stats manually and figure out what went wrong.
```

I know that I did some cherry-picking yesterday or the day before. My guess is something from my new rebalance code crept in.

I'll look now and try to fix for tonight's game.

**Comment by ctm** [comment]
Nothing changed in `tournament.rs`. My guess is this bug has been with us for a while and that it won't happen again soon. If so, it may be hard to track down.

**Comment by ctm** [comment]
Two players bust on on table 200 when we started out with 10 players. The tournament was 9-handed. My guess is the first bust triggered the breaking of the table and that somehow interfered with `remove_player` from being called for the second of the two busts.

If so, this should be fairly easy to reproduce, and it should also be fixed by the work I'm doing on table rebalancing (#10), but I don't want to wrap that work up until I absolutely know what happened here.

**Comment by ctm** [comment]
I was able to reproduce by starting a no-limit 6-handed game with 7 players. I then put 3 all-in and 2 bust. I then quit the remaining ones and sure enough, when the 2nd player quit, I got the ... pausing issue.

The pausing for more players is actually _relatively_ new, in that I added it on May 6th.

Anyway, since I now know how to reproduce it, I can go back to my work on #10, which won't be ready for this evening's game, but should be ready for tomorrow's.
